drivers/SmartThings/zigbee-window-treatment/src/aqara/init.lua

```diff
@@ -0,0 +1,258 @@
+local initializedstate = capabilities["stse.initializedstate"]
+local initializedStateId = "stse.initializedstate"
+local setInitializedStateCommandName = "setInitializedState"
+local opencloseDirectionPreferenceId = "stse.opencloseDirection"
+local softTouchPreferenceId = "stse.softTouch"
+
+local Basic = clusters.Basic
+local WindowCovering = clusters.WindowCovering
+local AnalogOutput = clusters.AnalogOutput
+local Groups = clusters.Groups
+local MFG_CODE = 0x115F
+local PREF_ATTRIBUTE_ID = 0x0401
+local INIT_STATE = "initState"
+local INIT_STATE_OPEN = "open"
+local INIT_STATE_CLOSE = "close"
+local INIT_STATE_REVERSE = "reverse"
+local INIT_STATE_DONE = "done"
+local SHADE_LEVEL = "shadeLevel"
+local SHADE_STATE = "shadeState"
+
+local FINGERPRINTS = {
+  { mfr = "LUMI", model = "lumi.curtain" },
+  { mfr = "LUMI", model = "lumi.curtain.v1" }
+}
+
+local is_aqara_products = function(opts, driver, device)
+  for _, fingerprint in ipairs(FINGERPRINTS) do
+      if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+          return true
+      end
+  end
+  return false
+end
+
+local read_custom_attribute = function(device, cluster_id, attribute)
+  local message = cluster_base.read_attribute(device, data_types.ClusterId(cluster_id), attribute)
+  message.body.zcl_header.frame_ctrl:set_mfg_specific()
+  message.body.zcl_header.mfg_code = data_types.validate_or_build_type(MFG_CODE, data_types.Uint16, "mfg_code")
+  return message
+end
+
+local read_pref_attribute = function(device)
+  device:send(read_custom_attribute(device, Basic.ID, PREF_ATTRIBUTE_ID))
+end
+
+local write_pref_attribute = function(device, str)
+  device:send(cluster_base.write_manufacturer_specific_attribute(device, Basic.ID, PREF_ATTRIBUTE_ID, MFG_CODE, data_types.CharString, str))
+end
+
+local function send_open_cmd(device, component)
+  device:send_to_component(component, WindowCovering.server.commands.GoToLiftPercentage(device, 100))
+end
+
+local function send_close_cmd(device, component)
+  device:send_to_component(component, WindowCovering.server.commands.GoToLiftPercentage(device, 0))
+end
+
+local do_refresh = function(self, device)
+  device:send(WindowCovering.attributes.CurrentPositionLiftPercentage:read(device))
+
+  read_pref_attribute(device)
+end
+
+local do_configure = function(self, device)
+  device:configure()
+
+  device:send(device_management.build_bind_request(device, WindowCovering.ID, self.environment_info.hub_zigbee_eui))
+
+  do_refresh(self, device)
+end
+
+local function device_added(driver, device)
+  local main_comp = device.profile.components["main"]
+  device:emit_component_event(main_comp,capabilities.windowShade.supportedWindowShadeCommands({"open", "close", "pause"}))
+  device:emit_component_event(main_comp,initializedstate.supportedInitializedState({"initialize"}))
+
+  write_pref_attribute(device, "\x00\x02\x00\x00\x00\x00\x00")
+  write_pref_attribute(device, "\x00\x08\x00\x00\x00\x00\x00")

```

Comment:
Do we have any documentation on what these values mean?  How are we intended to support this if we ever get reports that it's not working.  These seem to just be magic values.

Reply:
This is to write the default value of the curtain's preference menu as the initial settings of the curtain controller.